### PR TITLE
Admiral 41 acceleration movement actor

### LINKED ISSUE #41: Continuous Movement Actor Component
Include Continuous Movement Actor Component from Particle Env.

Movement is represented as velocity in x and y. These agents have a max speed. Velocity is damped, like friction, which we can treat like an entropy on the velocity. Velocity is updated by acceleration according to `dv = a * dt`. Position is then updated like `dp = v * dt`. `dt` is a scaling on the velocity and position updates, which we can normalize away.

Actions can be discrete UDLR, which translates to x, y acceleration vectors with magnitude = `move_speed`.
Actions can also directly be the x, y acceleration vector.

In Epic #39

### PR DESCRIPTION
**Pull request by rusu24edward**
Resolves #41

@glatt1 Here is an initial implementation of the velocity components. All you see here is the ability for agents to accelerate in 2 dimensions, which affects their velocities and thereby their positions. There is also the ability to add friction. I have not yet done anything with collisions since I'm going to handle that separately.

Take a look and let me know if anything seems off.

### CODE REVIEW

File: admiral/envs/components/actor.py [hunk 1]
```diff
@@ -230,6 +230,50 @@ def process_move(self, agent, acceleration, angle, **kwargs):
...
+    position_state (ContinuousPositionState):
+        The position state handler. Needed to modify agent positions.
+    
+    velocity_state (VelocityState):
+        The velocity state handler. Needed to modify agent velocities.
+    
+    agents (dict):
+        The dictionary of agents.
+    """
+    def __init__(self, position_state=None, velocity_state=None, agents=None, **kwargs):
```
Comment (glatt1) [inline admiral/envs/components/actor.py:247]: Thinking about space applications (also drones or submarines) we could add a dimension variable (standard = 2) and then use it in the action space. Wouldn't hurt here, I guess, and open up for potential cool applications.

File: admiral/envs/components/state.py [hunk 2]
```diff
@@ -245,6 +245,54 @@ def modify_ground_angle(self, agent, value, **kwargs):
...
+                else:
+                    agent.velocity = np.random.uniform(-agent.max_speed, agent.max_speed, (2,))
+    
+    def set_velocity(self, agent, _velocity, **kwargs):
+        """
+        Set the agent's velocity if it is within its max speed.
+        """
+        if isinstance(agent, VelocityAgent):
+            if np.linalg.norm(_velocity) < agent.max_speed:
+                agent.velocity = _velocity
```
Comment (glatt1) [inline admiral/envs/components/state.py:274]: Else it should be set to the max speed

File: admiral/envs/components/state.py [hunk 3]
```diff
@@ -245,6 +245,54 @@ def modify_ground_angle(self, agent, value, **kwargs):
...
+        """
+        Apply friction to the agent's movement, decreasing its speed by a small amount.
+        """
+        if isinstance(agent, VelocityAgent):
+            old_speed = np.linalg.norm(agent.velocity)
+            new_speed = old_speed - self.friction
+            if new_speed <= 0:
+                agent.velocity = np.zeros(2)
+            else:
+                agent.velocity *= new_speed / old_speed
```
Comment (glatt1) [inline admiral/envs/components/state.py:293]: if the old speed is zero, this fails as well

**Comment by glatt1** [comment]
Good, this looks ok. Only some small changes are necessary. Performance quickly gets slow if I add more agents but that's due to rendering so it doesn't matter much. I am curious in how this looks like when we include physics. 

100 agents:
rendering 5.882009029388428
no rendering 0.31136417388916016

1 agent:
rendering 5.520267009735107
no rendering 0.0039708614349365234

Additionally, I have two remarks that are not necessarily for this pull request:
1) I recommend making it directly 3D-ready but I realize it might be more difficult especially for the rendering part. Otherwise it wouldn't hurt to have a dimension variable that is standard on 2. But this is not a high priority so if it takes effort, I wouldn't do it now.
2) This doesn't really belong in this PR but it is a major issue: Every experiment must be reproducible or the environments have not much worth for research because we can't compare runs properly. There needs to be a random number generator which gets a seed for each environment and agent so that one can run identical experiments if they choose to. I would consider this a very high priority (outside of this PR). This is a huge thing in RL.

File: admiral/envs/components/actor.py (hunk 1 above)
Comment (rusu24edward) [reply admiral/envs/components/actor.py:247]: Captured in #46

File: admiral/envs/components/state.py (hunk 3 above)
Comment (rusu24edward) [reply admiral/envs/components/state.py:293]: If the old speed is 0 then the new speed will be less than or equal to zero, and that will be captured by the if block.

**Comment by rusu24edward** [comment]
>     2\. This doesn't really belong in this PR but it is a major issue: Every experiment must be reproducible or the environments have not much worth for research because we can't compare runs properly. There needs to be a random number generator which gets a seed for each environment and agent so that one can run identical experiments if they choose to. I would consider this a very high priority (outside of this PR). This is a huge thing in RL.

Captured in #47

**Pull request merged**